Storybook and sketchbook build

Mxchaeltrxn opened the pull request:
Three changes made:
1. Built the sketchbook config with webpack instead of esbuild, enabling styled-jsx. Styles are now applied (see the brandmark below).
![image](https://user-images.githubusercontent.com/34886045/130590934-b2e384df-e99b-46f3-a92c-ac91149b48c6.png)

2. Fixed styles on Github icon. in the header (vertically centered it).

3. Enable styles in Storybook by 'enabling' styled-jsx again via webpack (Github social link is now together, rather than apart like before).
![image](https://user-images.githubusercontent.com/34886045/130591262-975279c1-ede4-4190-9324-c098d91544b7.png)

haydn commented on .storybook/main.js:
```diff
@@ -1,4 +1,18 @@
+          loader: require("styled-jsx/webpack").loader,
+        },
+      ],
+    });
```
Soooo… this isn't doing anything. 😄 

The `test` option in this config tells Webpack to use this loader for files that match the given pattern. We don't have any CSS files that match the `/\.css$/` pattern, which means this loader is never used.

haydn commented on .storybook/.babelrc:
```diff
@@ -0,0 +1,12 @@
+        "sourceMaps": true
+      }
+    ],
+    "react-require"
```
This plugin isn't needed. Nowadays the `@babel/preset-react` has a `runtime` option that can be set to `"automatic"` to do this.

haydn commented on components/Stack.js:
```diff
@@ -19,6 +20,7 @@ const Stack = ({
         justify-content: ${justify};
         align-content: ${align};
         grid-auto-flow: ${DIRECTION_MAP[direction]};
+        align-items: ${alignItems};
```
This isn't needed. The reason things look like they're rendering incorrectly is because these global styles aren't getting loaded when we view a component in isolation:

https://github.com/sketchbook-js/website/blob/main/pages/index.js#L10-L65

The most obvious indicator of that is a serif font. This can be fixed in Storybook be configuring the story rendering:

https://storybook.js.org/docs/react/configure/overview#configure-story-rendering

haydn commented on .babelrc:
```diff
@@ -0,0 +1,12 @@
+    ],
+    "react-require"
+  ]
+}
```
We shouldn't need 2 Babel configs. One of them should be deleted (probably the `.storybook/.babelrc` one because it should fallback to using this one from its parent directory).

haydn commented on sketchbook/webpack.config.js:
```diff
@@ -0,0 +1,30 @@
+  },
+  resolve: {
+    extensions: ["*", ".js", ".jsx"],
+  },
```
This isn't needed. Webpack picks-up both `.js` and `.jsx` files by default.

haydn commented on sketchbook/webpack.config.js:
```diff
@@ -0,0 +1,30 @@
+            loader: require("styled-jsx/webpack").loader,
+          },
+        ],
+      },
```
As mentioned before, this isn't doing anything.

haydn reviewed (commented):
@Mxchaeltrxn Nice! You got both of them working. Sorry for the slow review.

I've added a bunch of comments cause there's a lot a little things here and there, but I've also fixed them all up — I'll push up the commits right after this.

Mxchaeltrxn commented:
@haydn Hey thanks for the review. 

You haven't yet merged so I'm thinking the remaining part of this PR is to deploy to the URL 
```
website-demo.sketchbookjs.com
```
Because
```
demo.sketchbookjs.com
```
Is already taken by the sketchbook repository.

Just checking since you didn't mention any other tasks in your comments.

haydn commented:
@Mxchaeltrxn Hello hello! Again, apologies for the slow response!

Yeah, let's setup a project in Vercel called `website-sketchbook` (to match the `website-storybook` project that already exists). Yep, let's use `website-demo.sketchbookjs.com` for now, it's super easy to change later if we want.

There are 2 changes I think we should make as well:

1.  Currently, the global CSS is duplicated between the `pages/index.js`, `.storybook/preview.js` and `sketchbook/config.jsx` files. It'd be better for those styles to be defined once in their own component and imported into those different locations. For the Sketchbook config it'll probably need to use [styled-jsx server-side rendering](https://github.com/vercel/styled-jsx#server-side-rendering).
2.  We can probably simplify the Storybook setup by removing the `@storybook/addon-essentials` addon. That means we won't have all actions, controls etc. For our purposes that's probably fine because we just want to demonstrate the components getting loaded into Stroybook. With that addon removed the stories definitions can be massively simplified so they don't have all that confusing args and template config.

If you'd like some inspiration, I've recently setup Storybook in this project I've been putting together for ColonyDB (it's a bit different because it's still using a few addons):

https://github.com/colonydb/anthill

I'm planning to add Sketchbook to that project soon too! 😄

haydn commented:
Actually… after thinking about it some more, I reckon you'll be able to get that global CSS into `sketchbook/config.jsx` using just `ReactDOM.render()`. Pretty sure something along these lines would work:

```js
const globalStylesDiv = doc.createElement("div");
doc.body.appendChild(globalStylesDiv);
ReactDOM.render(<GlobalCSSComponentThingy />, globalStylesDiv);
```

Mxchaeltrxn commented:
I'm still working on this so don't review it yet.

Mxchaeltrxn commented:
@haydn 
## Changes made
- [x] Put the duplicated CSS in one spot, much like you did in anthill.
- [x] Removed the storybook essentials addon and simplified the stories code.

It seems like there isn't an easy way for me to create a vercel project and test a production build before merging this to `main` branch so I'll just wait for that before I do that.

#### Question
On a semi-related note, should I be running `yarn sketchbook:build` and then trying to open `./sketchbook-build/index.html` and seeing if that loads Sketchbook to test the production build? I've been trying to do that locally but nothing seems to show up (I get a white screen), which makes me think that either something is wrong with the build or I'm not doing the right thing to load the build.

Also don't worry at all about your reply speed—I'm just glad you still review what I do!
